spec/services/push_notifications/send_spec.rb

```diff
@@ -91,4 +88,35 @@
       sidekiq_assert_enqueued_jobs(1, only: PushNotifications::DeliverWorker)
     end
   end
+
+  context "when associated to operational vs non-operational ConsumerApp" do
+    it "doesn't create any new (RPush) Push Notifications if non-operational" do
+      user3 = create(:user)
+      bad_consumer_app = create(:consumer_app, auth_key: nil)
+      create(:device, user: user3, consumer_app: bad_consumer_app)
+      bad_params = {
+        user_ids: [user3.id],
+        title: "Alert 3",
+        body: "some other alert!",
+        payload: ""
+      }
+
+      mocked_objects = mock_rpush(bad_consumer_app)
+
+      described_class.call(**bad_params)
+
+      # rpush_notification was saved 0 times -> No new notifications were created
+      expect(mocked_objects[:rpush_notification]).to have_received(:save!).exactly(0).times
```

Comment:
The bug was coming from the Rpush's Notification (in this case `Rpush::Apns2::Notification`). It was being created but the certificate wasn't available (in selfhost'ed Forem instances when users had used the iOS app). 

[This validation](https://github.com/rpush/rpush/blob/master/lib/rpush/client/active_model/apns2/app.rb#L9) was bubbling up an error all the way to the user when submitting a new comment. If the `ConsumerApp` isn't operational it means we don't need to create these records in the first place. The logic in place now takes this into account.

FYI these are the Redis-backed models that represent a (Push) Notification. Our own `Notification` model is indeed created in the DB. Not that this makes much of a difference, but worth clarifying it for context 🙂